Add pre-built binaries for `node-pty` package for Node.js v20

<!--- Please keep this note for the community --->

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

We currently only have pre-built binaries available for Node.js up until v19 ([source](https://github.com/cdktf/node-pty-prebuilt-multiarch/blob/9285d0e0cc3180832f49b67e6489bdd4c85fa512/.prebuild/buildify.js#L28)). As Node.js v20 has been released recently, we should add support for it.

Until we add support for this, users will need to have [dependencies](https://github.com/nodejs/node-gyp#installation) installed that allow the fallback of `node-gyp` to build that package. This e.g. includes a C/C++ compiler toolchain, Make, and Python.

<!--- Please leave a helpful description of the feature request here. --->

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

### References

This came up on the CDK Dev Slack with the following error message:

```
 > [base 4/5] RUN npm install -g cdktf-cli@0.15.4:
#12 31.58 npm WARN deprecated @npmcli/ci-detect@1.4.0: this package has been deprecated, use `ci-info` instead
#12 49.86 npm notice
#12 49.86 npm notice New patch version of npm available! 9.6.4 -> 9.6.5
#12 49.86 npm notice Changelog: <https://github.com/npm/cli/releases/tag/v9.6.5>
#12 49.87 npm notice Run `npm install -g npm@9.6.5` to update!
#12 49.87 npm notice
#12 49.89 npm ERR! code 1
#12 49.89 npm ERR! path /usr/local/lib/node_modules/cdktf-cli/node_modules/@cdktf/node-pty-prebuilt-multiarch
#12 49.90 npm ERR! command failed
#12 49.90 npm ERR! command sh -c node scripts/check-prebuild.js || prebuild-install || node scripts/install.js
#12 49.90 npm ERR! Prebuild binary missing for platform linux
#12 49.90 npm ERR! prebuild-install warn install No prebuilt binaries found (target=20.0.0 runtime=node arch=x64 libc=musl platform=linux)
#12 49.90 npm ERR! gyp info it worked if it ends with ok
#12 49.90 npm ERR! gyp info using node-gyp@9.3.1
#12 49.90 npm ERR! gyp info using node@20.0.0 | linux | x64
#12 49.91 npm ERR! gyp ERR! find Python
#12 49.91 npm ERR! gyp ERR! find Python Python is not set from command line or npm configuration
#12 49.91 npm ERR! gyp ERR! find Python Python is not set from environment variable PYTHON
#12 49.91 npm ERR! gyp ERR! find Python checking if "python3" can be used
#12 49.91 npm ERR! gyp ERR! find Python - "python3" is not in PATH or produced an error
#12 49.91 npm ERR! gyp ERR! find Python checking if "python" can be used
#12 49.91 npm ERR! gyp ERR! find Python - "python" is not in PATH or produced an error
#12 49.91 npm ERR! gyp ERR! find Python
#12 49.91 npm ERR! gyp ERR! find Python **********************************************************
#12 49.91 npm ERR! gyp ERR! find Python You need to install the latest version of Python.
#12 49.91 npm ERR! gyp ERR! find Python Node-gyp should be able to find and use Python. If not,
#12 49.91 npm ERR! gyp ERR! find Python you can try one of the following options:
#12 49.91 npm ERR! gyp ERR! find Python - Use the switch --python="/path/to/pythonexecutable"
#12 49.91 npm ERR! gyp ERR! find Python   (accepted by both node-gyp and npm)
#12 49.91 npm ERR! gyp ERR! find Python - Set the environment variable PYTHON
#12 49.91 npm ERR! gyp ERR! find Python - Set the npm configuration variable python:
#12 49.91 npm ERR! gyp ERR! find Python   npm config set python "/path/to/pythonexecutable"
#12 49.91 npm ERR! gyp ERR! find Python For more information consult the documentation at:
#12 49.91 npm ERR! gyp ERR! find Python https://github.com/nodejs/node-gyp#installation
#12 49.91 npm ERR! gyp ERR! find Python **********************************************************
#12 49.91 npm ERR! gyp ERR! find Python
#12 49.91 npm ERR! gyp ERR! configure error
#12 49.91 npm ERR! gyp ERR! stack Error: Could not find any Python installation to use
#12 49.91 npm ERR! gyp ERR! stack     at PythonFinder.fail (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:330:47)
#12 49.91 npm ERR! gyp ERR! stack     at PythonFinder.runChecks (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:159:21)
#12 49.91 npm ERR! gyp ERR! stack     at PythonFinder.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:202:16)
#12 49.91 npm ERR! gyp ERR! stack     at PythonFinder.execFileCallback (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:294:16)
#12 49.91 npm ERR! gyp ERR! stack     at exithandler (node:child_process:427:5)
#12 49.91 npm ERR! gyp ERR! stack     at ChildProcess.errorhandler (node:child_process:439:5)
#12 49.91 npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:511:28)
#12 49.91 npm ERR! gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:291:12)
#12 49.91 npm ERR! gyp ERR! stack     at onErrorNT (node:internal/child_process:483:16)
#12 49.91 npm ERR! gyp ERR! stack     at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
#12 49.91 npm ERR! gyp ERR! System Linux 5.10.104-linuxkit
#12 49.91 npm ERR! gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
#12 49.91 npm ERR! gyp ERR! cwd /usr/local/lib/node_modules/cdktf-cli/node_modules/@cdktf/node-pty-prebuilt-multiarch
#12 49.91 npm ERR! gyp ERR! node -v v20.0.0
#12 49.91 npm ERR! gyp ERR! node-gyp -v v9.3.1
#12 49.91 npm ERR! gyp ERR! not ok
```
<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor blog posts or documentation?
--->

--- a/packages/@cdktf/cli-core/package.json
+++ b/packages/@cdktf/cli-core/package.json
@@ -41,7 +41,7 @@
     "@cdktf/hcl2cdk": "0.0.0",
     "@cdktf/hcl2json": "0.0.0",
     "@cdktf/provider-schema": "0.0.0",
-    "@cdktf/node-pty-prebuilt-multiarch": "0.10.1-pre.10",
+    "@cdktf/node-pty-prebuilt-multiarch": "0.10.1-pre.11",
     "@sentry/node": "^7.64.0",
     "archiver": "^5.3.1",
     "cdktf": "0.0.0",

--- a/test/package.json
+++ b/test/package.json
@@ -13,7 +13,7 @@
   },
   "license": "MPL-2.0",
   "devDependencies": {
-    "@cdktf/node-pty-prebuilt-multiarch": "0.10.1-pre.10",
+    "@cdktf/node-pty-prebuilt-multiarch": "0.10.1-pre.11",
     "@skorfmann/terraform-cloud": "^1.9.1",
     "@types/express": "^4.17.13",
     "@types/fs-extra": "^8.1.0",

